src/lib.rs

```diff
@@ -794,10 +794,7 @@ pub unsafe extern "C" fn wgpuAdapterRequestDevice(
             return;
         }
     };
-    let base_limits = get_base_device_limits_from_adapter_limits(&adapter_limits)
-        // make sure we use the texture resolution limits from the adapter,
-        // so we can support images the size of the surface.
-        .using_resolution(adapter_limits);
```

Comment:
Should we still do *something* to at least support the surface size?

Reply:
> Should we still do _something_ to at least support the surface size?

The default limit is 8196, which seems like "good enough everywhere" for surfaces. The only reason you'd be using a lesser limit would be with downlevels, and then you simply can't support that big a texture anyway. This is however a terrifying thought so I'm curious as well.

Reply:
PS: I realize now the purpose of that line might be to make the size _smaller_ if necessary. In that case the code should choose the smaller default. But imo that should be handled by the base_limits function.

Reply:
> I realize now the purpose of that line might be to make the size smaller if necessary

Lets keep it per spec for now.

Reply:
The reason that's there in the wgpu examples is because we use the downlevel limits, which only requires tex size of 2k

Reply:
Now that I remember that's the same reason it was included here too, we determine _some_ "default" limits when user passes nothing, and it can be either of `downlevel_webgl2`/`downlevel`/`default.
So, when downlevel or lower is selected we need to use maximum texture size supported by the adapter, so that it can support display sizes more than 2K if the adapter supports it.

Reply:
The issue here (as stated in the issue) is that this gets applied unconditionally, so by default I'm getting limits much larger than the spec mandates.

Reply:
yeah, the fix here would be to clamp to webgpu spec default limits.